source/common/http/conn_pool_grid.cc

```diff
@@ -334,5 +334,58 @@ void ConnectivityGrid::onDrainReceived() {
   }
 }
 
+bool ConnectivityGrid::shouldAttemptHttp3() {
+  if (http3_status_tracker_.isHttp3Broken()) {
+    ENVOY_LOG(trace, "HTTP/3 is broken to host '{}', skipping.", host_->hostname());
+    return false;
+  }
+  if (!alternate_protocols_.has_value()) {
+    ENVOY_LOG(trace, "No alternate protocols cache. Attempting HTTP/3 to host '{}'.",
+              host_->hostname());
+    return true;
+  }
+  if (host_->address()->type() != Network::Address::Type::Ip) {
+    ENVOY_LOG(error, "Address is not an IP address");
+    ASSERT(false);
+    return false;
+  }
+  uint32_t port = host_->address()->ip()->port();
+  // TODO(RyanTheOptimist): Figure out how scheme gets plumbed in here.
```

Comment:
I think commenting that we only support secure QUIC regardless of url scheme would be fine.